src/profile/v100/json_data_convertible.rs

```diff
@@ -101,16 +93,31 @@ macro_rules! json_data_convertible {
 
                 #[test]
                 fn from_json_bytes_fail() {
-                    assert_eq!(
-                        SUT::new_from_json_bytes(BagOfBytes::sample()),
-                        Err(CommonError::FailedToDeserializeJSONToValue {
-                            json_byte_count: 32,
-                            type_name: {{
-                                const STRINGIFIED: &'static str = stringify!($type);
-                                STRINGIFIED
-                            }}.to_owned()
-                        })
-                    );
+                    let error: CommonError = SUT::new_from_json_bytes(BagOfBytes::sample()).unwrap_err();
+
+                    match error {
+                        CommonError::FailedToDeserializeJSONToValue {
+                            json_byte_count,
+                            type_name,
+                            ..
+                        } => {
+                            assert_eq!(
+                                32,
+                                json_byte_count,
+                            );
+
+                            assert_eq!(
+                                {{
+                                    const STRINGIFIED: &'static str = stringify!($type);
+                                    STRINGIFIED
+                                }}.to_owned(),
+                                type_name,
+                            );
+                        }
+                        _ => {
+                            error!("Expected CommonError::FailedToDeserializeJSONToValue but other error occurred")
```

Comment:
should probably `panic!` here instead of print an error. Most prints are filtered out, and when we `panic!` in a unit test, that is a failure of the unit test, which we want

Reply:
Yes that is what I was looking for. Confused with kotlin's error that does the same thing as panic here.